src/yenc.cc

```diff
@@ -204,6 +202,10 @@ PyObject* yenc_decode(PyObject* self, PyObject* Py_bytesarray_obj) {
         goto finish;
     }
 
+    // Create our destination bytearray
+    Py_output_bytearray = PyByteArray_FromStringAndSize(NULL, Py_buffer_obj->len);
```

Comment:
```suggestion
    Py_output_bytearray = PyByteArray_FromStringAndSize(NULL, yenc_data_length);
```

Should probably also guard against 0 length if that is possible by checking if PyByteArray_FromStringAndSize returns NULL